### PR TITLE
4 display data

### PR DESCRIPTION
**Pull request by gwan93**
Update as per PR comments: 

- Display legal watermarks for Map API
- The map's baselayer can be changed without a total re-render of the map
- Data is automatically reapplied when the baselayer is changed
- DRYing up repeated code

### CODE REVIEW

File: src/components/MapArea.js [hunk 1]
```diff
@@ -40,84 +50,86 @@ const trailData = {
...
 
   const changeStyle = (styleURL) => {
-    setStyle(styleURL);
-  }
+    if (map.current.getLayer("trailData")) {
+      map.current.removeLayer("trailData");
+    }
+    if (map.current.getSource("trailData")) {
+      map.current.removeSource("trailData");
+    }
```
Comment (parmvirthind) [inline src/components/MapArea.js]: does it not work without this?

File: src/components/MapArea.js [hunk 2]
```diff
@@ -40,84 +50,86 @@ const trailData = {
...
+    map.current.setStyle(styleURL);
+    map.current.on("style.load", () => {
+      displayDataOnMap(map, trailData);
+    });
```
Comment (parmvirthind) [inline src/components/MapArea.js]: since this is a map event hook, it should be defined when we initialize the map.

**Review by parmvirthind — CHANGES_REQUESTED** [review]
see comments

File: src/components/MapArea.js (hunk 2 above)
Comment (gwan93) [reply src/components/MapArea.js]: displayDataOnMap is also called once when the map is initialized. 106-108 is for when the baselayer style is changed. Line 105 calls map.current.setStyle(styleURL) which, from what I've read, takes some time to change the style. 106-108 will wait for the new style to finish applying before calling displayDataOnMap to reapply the source and layers. Without 106 and 108 then the data may not show, or errors such as "There is no source with this ID" when trying to remove the source prior to calling map.current.setStyle(styleURL)

File: src/components/MapArea.js (hunk 1 above)
Comment (gwan93) [reply src/components/MapArea.js]: You're correct, it will work. Will remove 99, 101, 102, and 104

File: src/components/MapArea.js (hunk 2 above)
Comment (parmvirthind) [reply src/components/MapArea.js]: It should work if you move the `style.load` event hook to the use effect where we define a new map.  We probably won't nee the `load` event hook anymore if `style.load` is being called when the baselayer is initially being set.

File: src/components/MapArea.js (hunk 1 above)
Comment (parmvirthind) [reply src/components/MapArea.js]: you can remove 99-104 and it should still work.

File: src/components/MapArea.js [hunk 3]
```diff
@@ -40,84 +50,78 @@ const trailData = {
...
-  });
+
+    return () => map.current.remove();
+  }, []);
 
   const changeStyle = (styleURL) => {
-    setStyle(styleURL);
-  }
+    map.current.setStyle(styleURL);
+    displayDataOnMap(map, trailData);
```
Comment (parmvirthind) [inline src/components/MapArea.js]: we have already estabilished a hook `style.load` that will be called everytime we change the style. So calling `displayDataOnMap` here is redundant.

**Pull request merged**